### PR TITLE
Mixed schemes gravity

### PR DESCRIPTION
**Pull request by alessiofumagalli**
The branch introduces the vector source term for both the rt0 and mvem discretizations. Moreover, the operator that created the reconstruction of the Darcy velocity is now a matrix and can be used in a more flexible way. Type hinting are added for the classes RT0, MVEM, and DualElliptic. Tests are added to check the vector source term.

### CODE REVIEW

File: src/porepy/numerics/fem/rt0.py [hunk 1]
```diff
@@ -15,13 +15,13 @@
 
 
 class RT0(pp.numerics.vem.dual_elliptic.DualElliptic):
-    def __init__(self, keyword):
+    def __init__(self, keyword: str):
```
Comment (keileg) [inline src/porepy/numerics/fem/rt0.py:18]:
```suggestion
    def __init__(self, keyword: str) -> None:
```

File: src/porepy/numerics/fem/rt0.py [hunk 2]
```diff
@@ -15,13 +15,13 @@
...
         super(RT0, self).__init__(keyword, "RT0")
         # variable name to store the structure that map a cell to the opposite nodes
         # of the local faces
         self.cell_face_to_opposite_node = "rt0_class_cell_face_to_opposite_node"
 
-    def discretize(self, g, data):
+    def discretize(self, g: pp.Grid, data: dict):
```
Comment (keileg) [inline src/porepy/numerics/fem/rt0.py:24]:
```suggestion
    def discretize(self, g: pp.Grid, data: Dict) -> None:
```

Comment (keileg) [reply src/porepy/numerics/fem/rt0.py:24]: You will also need from typing import Dict.

File: src/porepy/numerics/fem/rt0.py [hunk 3]
```diff
@@ -57,9 +57,9 @@ def discretize(self, g, data):
         matrix_dictionary = data[pp.DISCRETIZATION_MATRICES][self.keyword]
         # If a 0-d grid is given then we return an identity matrix
         if g.dim == 0:
-            mass = sps.dia_matrix(([1], 0), (g.num_faces, g.num_faces))
-            matrix_dictionary["mass"] = mass
-            matrix_dictionary["div"] = sps.csr_matrix((g.num_faces, g.num_cells))
+            matrix_dictionary[self.mass_matrix_key] = sps.dia_matrix(([1], 0), (1, 1))
```
Comment (keileg) [inline src/porepy/numerics/fem/rt0.py:60]: Very good!

File: src/porepy/numerics/fem/rt0.py [hunk 4]
```diff
@@ -74,9 +74,10 @@ def discretize(self, g, data):
         # Map the domain to a reference geometry (i.e. equivalent to compute
         # surface coordinates in 1d and 2d)
         deviation_from_plane_tol = data.get("deviation_from_plane_tol", 1e-5)
```
Comment (keileg) [inline src/porepy/numerics/fem/rt0.py:76]: As a general comment, I think this is a rather loose tolerance for deviations; in practice we need to be much stricter not to get problems in the meshing. But that probably is an argument that the test is not much needed.

Another issues is that it is stupid not to have a relative tolerance here, but that is common to all computational geometry in pp, and it is my bad.

File: src/porepy/numerics/vem/dual_elliptic.py [hunk 5]
```diff
@@ -66,11 +67,23 @@ class DualElliptic(
 
     """
 
-    def __init__(self, keyword, name):
+    def __init__(self, keyword: str, name: str):
```
Comment (keileg) [inline src/porepy/numerics/vem/dual_elliptic.py:70]:
```suggestion
    def __init__(self, keyword: str, name: str) -> None:
```

File: src/porepy/numerics/vem/mvem.py [hunk 6]
```diff
@@ -22,10 +23,10 @@ class MVEM(pp.numerics.vem.dual_elliptic.DualElliptic):
 
     """
 
-    def __init__(self, keyword):
+    def __init__(self, keyword: str):
         super(MVEM, self).__init__(keyword, "MVEM")
 
-    def discretize(self, g, data):
+    def discretize(self, g: pp.Grid, data: dict):
```
Comment (keileg) [inline src/porepy/numerics/vem/mvem.py:29]:
```suggestion
    def discretize(self, g: pp.Grid, data: Dict) -> None:
```

File: src/porepy/numerics/vem/mvem.py [hunk 7]
```diff
@@ -22,10 +23,10 @@ class MVEM(pp.numerics.vem.dual_elliptic.DualElliptic):
 
     """
 
-    def __init__(self, keyword):
+    def __init__(self, keyword: str):
```
Comment (keileg) [inline src/porepy/numerics/vem/mvem.py:26]:
```suggestion
    def __init__(self, keyword: str) -> None:
```

File: src/porepy/numerics/fem/rt0.py (hunk 4 above)
Comment (alessiofumagalli) [reply src/porepy/numerics/fem/rt0.py:76]: yes, I agree with you..

**Pull request merged**